scripts/pre_commit_linter.py

```diff
@@ -116,22 +122,66 @@ def _get_changed_filenames():
     return unstaged_files + staged_files
 
 
-def _get_all_files_in_directory(dir_path):
+def _is_filename_in_glob_patterns(filename, glob_patterns):
+    """Checks whether 'filename' by any pattern in 'glob_patterns'.
+
+    Args:
+    - filename: str. Filename to check
+    - glob_patterns: list. List of glob_patterns to check filename against
+
+    Returns:
+        bool: whether filename matches any glob_patterns.
+
+    >>> _is_filename_in_glob_patterns('foo/bar', ['foo/', 'bar/*'])
+    False
+    >>> _is_filename_in_glob_patterns('foo/bar', ['foo/foo', 'foo/**'])
+    True
+    >>> _is_filename_in_glob_patterns('bar/bar', ['bar/bar', 'foo/bar'])
+    True
+    """
+    for glob_pattern in glob_patterns:
+        if fnmatch.fnmatch(filename, glob_pattern):
+            return True
+    return False
+
+
+def _get_glob_patterns_excluded_from_jscsrc(config_jscsrc):
+    """Collects excludeFiles from jscsrc file.
+
+    Args:
+    - config_jscsrc: str. Path to .jscsrc file.
+
+    Returns:
+        a list of files in excludeFiles.
+    """
+    with open(config_jscsrc) as f:
+        f.readline()  # First three lines are comments
+        f.readline()
+        f.readline()
+        json_data = json.loads(f.read())
+
+    return json_data['excludeFiles']
+
+
+def _get_all_files_in_directory(dir_path, excluded_glob_patterns):
     """Recursively collects all files in directory and
     subdirectories of specified path.
 
     Args:
     - dir_path: str. Path to the folder to be linted.
+    - excluded_glob_patterns: set. Set of all files to be excluded.
 
     Returns:
-        a list of files in directory and subdirectories.
+        a list of files in directory and subdirectories without excluded files.
     """
     files_in_directory = []
     for _dir, _, files in os.walk(dir_path):
         for file_name in files:
-            files_in_directory.append(
-                os.path.relpath(os.path.join(_dir, file_name), os.getcwd()))
-
+            filename = os.path.relpath(os.path.join(_dir, file_name),
+                                       os.getcwd())
+            if not _is_filename_in_glob_patterns(filename,
```

Comment:
Optional: you could replace this with a one-liner:

   if not any([fnmatch.fnmatch(filename, gp) for gp in excluded_glob_patterns]):
       files_in_directory.append(filename)

and then remove the entirety of _is_filename_in_glob_patterns.